### ISSUE TITLE
Cannot SET XACT_ABORT

### ISSUE THREAD

**Issue by UFTimmy** [Microsoft/msphpsql#690]
**+## Driver version or file name**
13.1.9.2-1

 **+## SQL Server version**
Microsoft SQL Server 2014 (SP2-GDR) (KB3194714) - 12.0.5203.0 (X64) 
	Sep 23 2016 18:13:56 
	Copyright (c) Microsoft Corporation
	Developer Edition (64-bit) on Windows NT 6.3 <X64> (Build 9600: ) (Hypervisor)

 **+## Client operating system**
Ubuntu 16.04

 **+## PHP version**
7.2.1

 **+## Problem description**
Unable to turn on XACT_ABORT to perform a distributed transaction.

 **+## Expected behavior and actual behavior**
I should be able to SET XACT_ABORT ON, begin a transaction, and have the setting work.

 **+## Repro code**
I need to perform a transaction on both a local table, and on a linked server. Using the provided sqlsrv_begin_transaction does not seem to honor the SET XACT_ABORT ON statement.

In other words, this does not work:
sqlsrv_query(SET XACT_ABORT ON)
sqlsrv_begin_transaction();
UPDATE local table
UPDATE linked table
sqlsrv_commit()

It fails at the UPDATED linked table with:
SQLSTATE [42000, 7395]: [Microsoft][ODBC Driver 13 for SQL Server][SQL Server]Unable to start a nested transaction for OLE DB provider "SQLNCLI11" for linked server "XXX". A nested transaction was required because the XACT_ABORT option was set to OFF.

SQLSTATE [01000, 7412]: [Microsoft][ODBC Driver 13 for SQL Server][SQL Server]OLE DB provider "SQLNCLI11" for linked server "XXX" returned message "Cannot start more transactions on this session.". 

However, if I begin a transaction manually through sqlsrv_query, it works:
sqlsrv_query(SET XACT_ABORT ON)
sqlsrv_query(BEGIN TRANSACTION)
UPDATE local table
UPDATE linked table
sqlsrv_query(COMMIT TRANSACTION)

**Is there a way to turn on XACT_ABORT and have it be recognized by the sqlsrv_begin_transaction function?**

**Comment by yitam** [comment]
Hello @UFTimmy 

In general, we do not recommend mixing [Transact-SQL](https://docs.microsoft.com/en-us/sql/t-sql/language-elements/transactions-transact-sql) programming with sqlsrv transaction related calls, mainly because transactions are handled differently. 

If you prefer T-SQL programming, you may consider using stored procedure(s) instead. Note that when `sqlsrv_begin_transaction()` is used, you will need to programmatically check for error cases to roll back the transaction when errors occur.

Hope this clarifies things.

**Comment by UFTimmy** [comment]
Sorry, no, it doesn't really help me at all. Can you clarify?

> In general, we do not recommend mixing Transact-SQL programming with sqlsrv transaction related calls, 

In this case, which part is Transact-SQL programming and which part is sqlsrv transaction related?

I don't want to mix anything, but I need to call SET XACT_ABORT ON, and it does not seem to work when doing transactions the recommend way. So how can I do that?

**Comment by yitam** [comment]
Hi @UFTimmy, the answer to your question "Is there a way to turn on XACT_ABORT and have it be recognized by the sqlsrv_begin_transaction function?" is 

No, it is not possible. 

FYI, T-SQL (Transact-SQL) is a set of programming extensions from Sybase and Microsoft that add several features to the Structured Query Language (SQL), including transaction control, exception and error handling, row processing and declared variables.

In other words, SET XACT_ABORT ON is one T-SQL way to control how transactions work, and sqlsrv_begin_transaction() is another. Mixing these two mechanisms is not recommended. As you have already observed, it simply doesn't work.

If you really prefer using SET XACT_ABORT ON, I'd suggest you design a stored procedure for `sqlsrv_query()` to invoke.

**Comment by yitam** [comment]
Closing this issue due to inactivity. Please feel free to reopen if you still have other questions.
